### PR TITLE
Feature give committees access to own ticket sales

### PR DESCRIPTION
**Pull request by julian9499**
solves #319

### CODE REVIEW

File: src/main/resources/templates/sales/stats/products/index.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                                <div class="col">
+                                    <div class="card mb-3">
+                                        <h6 class="card-header">Money earned for this product:</h6>
+                                        <div class="card-body text-primary">
+                                            <h4 class="card-title">
+                                                € [[${product.getCost() * product.getSold()}]]
```
Comment (Fastjur) [inline src/main/resources/templates/sales/stats/products/index.html]: Maybe delimit this to 2 decimals

File: src/main/resources/templates/sales/stats/products/index.html [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                </div>
+            </div>
+        </main>
+    </div>
+</div>
+</body>
+
+<script>
+    document.addEventListener("click", redirect);
+    document.addEventListener("touchstart", redirect);
```
Comment (Fastjur) [inline src/main/resources/templates/sales/stats/products/index.html]: Same here, what is this?

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsMainController.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * @param authenticationService of type AuthenticationService.
+     * @param salesService          of type SalesService.
+     */
+    public SalesStatsMainController(AuthenticationService authenticationService, SalesService salesService) {
+        this.authenticationService = authenticationService;
+        this.salesService = salesService;
+    }
+
+    /**
+     * Index view of the scan app.
```
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsMainController.java]: Fixed!

File: .codeclimate/checkstyle.xml [hunk 4]
```diff
@@ -71,6 +72,12 @@
         <property name="message" value="Line has trailing spaces."/>
     </module>
 
+    <!-- Checks for Size Violations.                    -->
+    <!-- See http://checkstyle.sf.net/config_sizes.html -->
+    <module name="LineLength">
```
[the remark this answers is not in the document]
Comment (julian9499) [reply .codeclimate/checkstyle.xml]: Checkstyle errors. Put it back

File: src/main/resources/templates/admin/orders/index.html [hunk 5]
```diff
@@ -51,8 +51,8 @@ <h1>Orders</h1>
                         <td th:text="${#temporals.format(order.getCreatedAt(), 'yyyy-MM-dd')}"></td>
                         <td><samp th:text="${order.getPublicReference().split('-')[0]}"></samp></td>
                         <td>
-                            <a th:href="@{'/administrator/customers/view/' + ${order.getOwner().getKey()} + '/'}"
-                               th:text="${order.getOwner().getName()}" th:if="${order.getOwner() != null}"></a>
+                            <span th:href="@{'/administrator/customers/view/' + ${order.getOwner().getKey()} + '/'}"
```
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/resources/templates/admin/orders/index.html]: Fixed!

File: src/main/resources/templates/sales/index.html [hunk 6]
```diff
@@ -32,13 +32,19 @@ <h4 class="">Menu</h4>
...
                     <div class="row justify-content-between">
-                        <div class="col-6 text-center">
+                        <div class="col-4 text-center">
                             <a th:href="@{/sales/sell}">
                                 <i class="fa fa-ticket-alt fa-fw fa-4x"></i>
                                 <p>Sell tickets</p>
                             </a>
                         </div>
-                        <div class="col-6 text-center">
+                        <div class="col-4 text-center">
```
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/resources/templates/sales/index.html]: There were 2 buttons here, but now three. So they had to be changed to col-4.

File: src/main/resources/templates/sales/stats/products/index.html (hunk 1 above)
Comment (julian9499) [reply src/main/resources/templates/sales/stats/products/index.html]: No

File: src/main/resources/templates/sales/stats/products/index.html (hunk 2 above)
Comment (julian9499) [reply src/main/resources/templates/sales/stats/products/index.html]: Fixed!

File: src/main/resources/templates/sales/stats/event/index.html [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                    </table>
+                </div>
+        </main>
+    </div>
+</div>
+</body>
+
+<script>
+    document.addEventListener("click", redirect);
+    document.addEventListener("touchstart", redirect);
```
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/resources/templates/sales/stats/event/index.html]: Fixed!

File: src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java [hunk 8]
```diff
@@ -61,4 +73,22 @@ public SalesServiceImpl(EventService eventService) {
                 .filter(product -> LocalDateTime.now().isAfter(product.getSellStart()) && LocalDateTime.now().isBefore(product.getSellEnd()))
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public List<Order> getAllOrdersByEvent(Event event) throws EventNotFoundException {
+        List<Order> orders = orderService.getAllOrders();
```
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java]: Jpa is not helping me here 
Does someone know how to fix this?

File: .codeclimate/checkstyle.xml [hunk 9]
```diff
@@ -30,6 +30,7 @@
 
 -->
 
+
```
Comment (skamoen) [inline .codeclimate/checkstyle.xml:33]: Yay more whitespace

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+ * SalesScanEventController.
+ */
+@Controller
+@RequestMapping(value = "/sales/stats/event/{key}")
+@PreAuthorize("hasRole('USER')")
+public class SalesStatsEventController {
+    /** Default return redirect on error. */
+    private static final String ERROR_REDIRECT = "redirect:/sales/stats/";
+
+    /** OrderService. */
```
Comment (skamoen) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java:29]: You're not making anyone happy with comments like this... Just remove 'm if they don't add anything

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+     * @return String
+     */
+    @GetMapping
+    public String index(Model model, @PathVariable String key) {
+        Event event;
+        List<Order> orders;
+        try {
+            event = eventService.getByKey(key);
+            orders = salesService.getAllOrdersByEvent(event);
+        } catch(EventNotFoundException e){
```
Comment (skamoen) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java:67]: Maybe catch this using a ExceptionHandler? See here for an example: https://github.com/AreaFiftyLAN/lancie-api/blob/master/src/main/java/ch/wisv/areafiftylan/users/controller/UserRestController.java#L188-L191

Any exception thrown within your controller will use that to decide what to do.

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsMainController.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+    /**
+     * Index view of the stats page.
+     *
+     * @param model of type Model.
+     * @return String
+     */
+    @GetMapping
+    public String index(Model model) {
+        Customer currentUser = authenticationService.getCurrentCustomer();
```
Comment (skamoen) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsMainController.java:49]: You could alternatively add this as a method parameter and it will be autowired if it uses the Spring Security Context. This works too though

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsProductController.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+/**
+ * SalesScanEventController.
+ */
+@Controller
+@RequestMapping(value = "/sales/stats/products/{key}")
```
Comment (skamoen) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsProductController.java:17]: If you're going to define only one endpoint per controller, you can also just merge them and define the mapping per method. You can still define a "parent" endpoint `/sales/stats` for all methods in the combined controller.

See https://github.com/AreaFiftyLAN/lancie-api/blob/master/src/main/java/ch/wisv/areafiftylan/users/controller/UserRestController.java#L46 for an example

File: src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (skamoen) [reply src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java:79]: This doesn't look resolved to me

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsEventControllerTest.java [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+/**
+ * SalesScanMainControllerTest.
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = EventsApplicationTest.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test")
+public class SalesStatsEventControllerTest extends ControllerTest {
```
Comment (skamoen) [inline src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsEventControllerTest.java:26]: Just put all these tests in 1 file

File: src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java:79]: We tried to go through soldproduct right after the initial comment, but it is not possible. At least I and @Fastjur could not figure it out

[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java]: I've linked it via another way

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java (hunk 10 above)
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java]: done

File: .codeclimate/checkstyle.xml (hunk 9 above)
Comment (julian9499) [reply .codeclimate/checkstyle.xml]: done

File: src/main/java/ch/wisv/events/core/model/event/Event.java [hunk 15]
```diff
@@ -227,4 +227,13 @@ public int getReserved() {
     private double calcProgress(double reserved) {
         return Math.min(Math.round(((reserved / (double) this.target) * 100.d) * 100.d) / 100.d, 100.d);
     }
+
+    /**
+     * Get all products of this event
+     *
+     * @return List<Product>
+     */
+    public List<Product> getProducts(){
```
Comment (skamoen) [inline src/main/java/ch/wisv/events/core/model/event/Event.java:236]: This class has a `@Data` annotation, so boilerplate getters like this are already generated

Comment (julian9499) [reply src/main/java/ch/wisv/events/core/model/event/Event.java]: Removed

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java (hunk 11 above)
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java]: Done

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsProductController.java (hunk 13 above)
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsProductController.java]: Done

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsMainController.java (hunk 12 above)
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsMainController.java]: I tried this, but I can only get the authentication. I could match the authentication with the customer in the function itself, but I think the current location of the getCurrentCustomer function is better placed in the authenticationService.

File: src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java [hunk 16]
```diff
@@ -302,6 +303,18 @@ public boolean containsOnlyReservable(Order order) {
         );
     }
 
+    /**
+     * Returns a list of orders based on product
+     */
+    @Override
+    public List<Order> getAllByProduct(Product product){
+        List<Order> orders = new ArrayList<>();
+        for ( OrderProduct orderProduct :orderProductRepository.findAllByProduct(product)){
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java:312]: Fix formatting

File: src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java [hunk 17]
```diff
@@ -302,6 +303,18 @@ public boolean containsOnlyReservable(Order order) {
...
+            orders.addAll(orderRepository.findAllByOrderProducts(orderProduct));
+        }
+        return orders;
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java:315]: enter before return

File: src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java (hunk 16 above)
Comment (svenpopping) [inline src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java:312]: enter before for-loop

Comment (svenpopping) [reply src/main/java/ch/wisv/events/core/service/order/OrderServiceImpl.java:312]: or replace with a lambda function, as it seems perfect for this job.

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    /**
+     * Index view of the stats page.
+     *
+     * @param model of type Model.
+     * @return String
+     */
+    @GetMapping
+    public String IndexView(Model model) {
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:72]: method name should not start with capital.

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java [hunk 19]
```diff
@@ -0,0 +1,129 @@
...
+        model.addAttribute("events", salesService.getAllGrantedEventByCustomer(currentUser));
+
+        return "sales/stats/index";
+    }
+
+    /**
+     * Ticket sales view.
+     *
+     * @param model of type Model
+     * @param key    of type String
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:83]: fix formatting

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java [hunk 20]
```diff
@@ -0,0 +1,129 @@
...
+        model.addAttribute("key", key);
+
+        return "sales/stats/products/index";
+    }
+
+    /**
+     * Event sales view.
+     *
+     * @param model of type Model
+     * @param key    of type String
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:100]: fix formatting

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+        for(Product product: event.getProducts()){
+            moneyEarned += product.getCost() * product.getSold();
+            ticketsSold += product.getSold();
+        }
+
+        model.addAttribute("orders", orders);
+        model.addAttribute("moneyEarned", moneyEarned);
+        model.addAttribute("ticketsSold", ticketsSold);
+        model.addAttribute("event", event);
+        model.addAttribute("key", key);
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:119]: can be remove and replaced by `event.getKey()` in the frontend.

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+        List<Order> orders = salesService.getAllOrdersByEvent(event);
+        double moneyEarned = 0.0;
+        int ticketsSold = 0;
+
+        for(Product product: event.getProducts()){
+            moneyEarned += product.getCost() * product.getSold();
+            ticketsSold += product.getSold();
+        }
+
+        model.addAttribute("orders", orders);
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:115]: The orders should not be added to the frontend as now it leaking additional information about the orders. For example, the information about other tickets the customer bought.
The order information is not relevant for the committee members, the names of the participants might be useful.

File: src/main/java/ch/wisv/events/sales/service/SalesService.java [hunk 23]
```diff
@@ -20,4 +22,11 @@
      * @return List of Products
      */
     List<Product> getAllGrantedProductByCustomer(Customer customer);
+
+    /**
+     * Get all orders of an event
+     * @param event
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/service/SalesService.java:28]: fix formatting

File: src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java [hunk 24]
```diff
@@ -61,4 +73,15 @@ public SalesServiceImpl(EventService eventService) {
                 .filter(product -> LocalDateTime.now().isAfter(product.getSellStart()) && LocalDateTime.now().isBefore(product.getSellEnd()))
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public List<Order> getAllOrdersByEvent(Event event) {
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java:78]: Add javadoc

File: src/main/java/ch/wisv/events/core/service/order/OrderService.java [hunk 25]
```diff
@@ -104,6 +106,12 @@
      */
     List<Order> getAllPaid();
 
+    /**
+     * Get all Orders by product
+     * @return List of Orders
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/core/service/order/OrderService.java:111]: Fix formatting

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java [hunk 26]
```diff
@@ -0,0 +1,114 @@
...
+        when(salesService.getAllOrdersByEvent(event)).thenReturn(new ArrayList<>());
+
+        mockMvc.perform(get("/sales/stats/products/" + event.getKey()))
+                .andExpect(status().isOk())
+                .andExpect(view().name("sales/stats/products/index"))
+                .andExpect(model().attributeExists("products", "key"))
+                .andExpect(model().attribute("products", is(event.getProducts())))
+                .andExpect(model().attribute("key", is(event.getKey())));
+    }
+}
```
Comment (svenpopping) [inline src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java:114]: newline

File: src/main/resources/templates/sales/stats/sidebar.html [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+<nav class="col-sm-3 col-md-2 d-none d-sm-block sidebar" th:fragment="sidebar">
+    <ul class="nav nav-pills flex-column">
+        <li class="nav-item">
+            <a th:href="@{'/sales/stats/event/' + ${key}}" class="nav-link"
+               th:classappend="${#httpServletRequest.getRequestURI().contains('/sales/stats/event/') ? 'active':''}">
+                <i class="fa fa-calendar fa-fw" aria-hidden="true"></i> Event
+            </a>
+        </li>
+        <li class="nav-item">
+            <a th:href="@{'/sales/stats/products/'+${key}}" class="nav-link"
```
Comment (svenpopping) [inline src/main/resources/templates/sales/stats/sidebar.html:16]: Fix formatting

Comment (svenpopping) [reply src/main/resources/templates/sales/stats/sidebar.html:16]: replace `${key}` by `${event.getKey()}`

File: src/main/resources/templates/sales/stats/sidebar.html [hunk 28]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org">
+<head></head>
+
+<body>
+<!--/*@thymesVar id="key" type="java.lang.String"*/-->
+<nav class="col-sm-3 col-md-2 d-none d-sm-block sidebar" th:fragment="sidebar">
+    <ul class="nav nav-pills flex-column">
+        <li class="nav-item">
+            <a th:href="@{'/sales/stats/event/' + ${key}}" class="nav-link"
```
Comment (svenpopping) [inline src/main/resources/templates/sales/stats/sidebar.html:10]: replace `${key}` by `${event.getKey()}`, to make it clear which type of key it is.

File: src/main/resources/templates/sales/stats/products/index.html [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+    <link rel="stylesheet" th:href="@{/webjars/font-awesome/5.0.13/web-fonts-with-css/css/fontawesome-all.min.css}">
+</head>
+
+<body style="width: calc(100vw); height: calc(100vh)">
+<nav th:replace="/sales/fragments/header :: header"></nav>
+<div class="container-fluid">
+    <div class="row">
+        <nav th:replace="/sales/stats/sidebar :: sidebar"></nav>
+        <main class="col-sm-9 ml-sm-auto col-md-10 p-5" role="main">
+            <div class="row" th:each=" product : ${products}">
```
Comment (svenpopping) [inline src/main/resources/templates/sales/stats/products/index.html:28]: fix formatting

File: src/main/resources/templates/sales/stats/products/index.html [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+                                            <h4 class="card-title">
+                                                <small>no.</small>
+                                                [[${product.getSold()}]]
+                                                /
+                                                [[${product.getMaxSold()}]]
+                                            </h4>
+                                            <div class="progress mb-2">
+                                                <div class="progress-bar" role="progressbar" aria-valuemin="0"
+                                                     aria-valuemax="100"
+                                                     th:style="'width: ' + ${((product.getSold()*1.0)/(product.getMaxSold()*1.0)) * 100.0} + '%'">
```
Comment (svenpopping) [inline src/main/resources/templates/sales/stats/products/index.html:47]: fix formatting around `/`

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java [hunk 31]
```diff
@@ -0,0 +1,114 @@
...
+    private SalesService salesService;
+
+    /**
+     * Test stats event controller
+     *
+     * @throws Exception when exception
+     */
+    @Test
+    public void eventsControllerTest() throws Exception {
+
```
Comment (svenpopping) [inline src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java:51]: remove

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java [hunk 32]
```diff
@@ -0,0 +1,114 @@
...
+                LocalDateTime.now().plusHours(1),
+                "Short description"
+        );
+
+        when(eventService.getByKey(event.getKey())).thenReturn(event);
+
+        mockMvc.perform(get("/sales/stats/event/" + event.getKey()))
+                .andExpect(status().isOk())
+                .andExpect(view().name("sales/stats/event/index"))
+                .andExpect(model().attributeExists("event", "moneyEarned", "ticketsSold", "orders", "key"));
```
Comment (svenpopping) [inline src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java:69]: remove `key`

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java [hunk 33]
```diff
@@ -0,0 +1,114 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+/**
+ * SalesScanMainControllerTest.
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = EventsApplicationTest.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test")
+public class SalesStatsControllersTest extends ControllerTest {
```
Comment (svenpopping) [inline src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java:29]: Contains no test including orders or edge cases. Please add them

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java [hunk 34]
```diff
@@ -0,0 +1,129 @@
...
+     * @return String
+     */
+    @GetMapping("/event/{key}")
+    public String eventSalesView(Model model, @PathVariable String key) throws EventNotFoundException{
+        Event event = eventService.getByKey(key);
+        List<Order> orders = salesService.getAllOrdersByEvent(event);
+        double moneyEarned = 0.0;
+        int ticketsSold = 0;
+
+        for(Product product: event.getProducts()){
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:110]: fix formatting

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java [hunk 35]
```diff
@@ -0,0 +1,114 @@
...
+                "test",
+                10,
+                10,
+                "path/to/files",
+                LocalDateTime.now(),
+                LocalDateTime.now().plusHours(1),
+                "Short description"
+        );
+
+        when(eventService.getByKey(event.getKey())).thenReturn(event);
```
Comment (svenpopping) [inline src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java:64]: Make it an integration test by creating models and save them and removing the mocking (for most models a method in `ControllerTest` has been created). 
This test method is quite useless now.
In this cases your test cases do not test the added methods to the `OrderService`, `OrderProductRepo`, etc.

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java [hunk 36]
```diff
@@ -0,0 +1,114 @@
...
```
Comment (svenpopping) [inline src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java:104]: Same as above

**Review by svenpopping — DISMISSED** [review]

File: src/main/resources/templates/sales/stats/event/index.html [hunk 37]
```diff
@@ -0,0 +1,93 @@
...
+                            <td th:text="${order.getStatus()}"></td>
+                        </tr>
+                        </tbody>
+                    </table>
+                </div>
+        </main>
+    </div>
+</div>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/resources/templates/sales/stats/event/index.html:93]: done

File: src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java [hunk 38]
```diff
@@ -61,4 +73,22 @@ public SalesServiceImpl(EventService eventService) {
...
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public List<Order> getAllOrdersByEvent(Event event) throws EventNotFoundException {
+        List<Order> orders = orderService.getAllOrders();
+
+        List<Order> ordersAssociatedWithEvent = new ArrayList<>();
+
+        for (Order order : orders) {
```
[the remark this answers is not in the document]
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/service/SalesServiceImpl.java]: Done

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java (hunk 32 above)
Comment (julian9499) [reply src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java:37]: Done

File: src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java (hunk 35 above)
Comment (julian9499) [reply src/test/java/ch/wisv/events/sales/controller/stats/SalesStatsControllersTest.java]: Changed it to use the models found in `ControllerTest`. I will write tests for OrderService and OrderProductRepo next.

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java (hunk 34 above)
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:110]: Done

**Comment by Fastjur** [comment]
I keep getting pinged about this one 😄 

What is the status? It's pretty old but still a very useful feature IMO!

**Comment by julian9499** [comment]
The feature was currently on a lower priority due to the lesser amount of activities that are organisted by CH, but I've got some free time next week which I will use to apply sven's suggestions to the code base which hopefully results in us being able to merge this and other features. I do have a problem with the codeclimate runner for checkstyle. It states that our checkstyle config is invalid and I have not been able to solve it.

**Comment by Fastjur** [comment]
I understand. No rush, was just wondering :)

I have time coming weekend or that week after, so I can review then and if need be help with the codeclimate issue. I cannot see any error messages from it for some reason.

File: src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java (hunk 22 above)
Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:117]: The order is currently used for a few parameters:
```
publicReference
createdAtDate
orderProducts
getAmount
getStatus
```
I will set all data that should not be available to the front-end to null to allow the data to be anonymously used

Comment (julian9499) [reply src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java:117]: Is this sufficient?

**Review by Fastjur — APPROVED** [review]
